yarn-project/pxe/src/database/pxe_database.ts

```diff
@@ -209,18 +208,29 @@ export interface PxeDatabase extends ContractArtifactDatabase, ContractInstanceD
 
   /**
    * Returns the last seen indexes for the provided app siloed tagging secrets or 0 if they've never been seen.
-   * The recipient must also be provided to convey "directionality" of the secret and index pair, or in other words
-   * whether the index was used to tag a sent or received note.
    * @param appTaggingSecrets - The app siloed tagging secrets.
    * @returns The indexes for the provided secrets, 0 if they've never been seen.
    */
-  getTaggingSecretsIndexes(appTaggingSecretsWithRecipient: TaggingSecret[]): Promise<number[]>;
+  getTaggingSecretsIndexesAsRecipient(appTaggingSecrets: Fr[]): Promise<number[]>;
 
   /**
-   * Increments the index for the provided app siloed tagging secrets.
-   * The recipient must also be provided to convey "directionality" of the secret and index pair, or in other words
-   * whether the index was used to tag a sent or received note.
+   * Returns the last seen indexes for the provided app siloed tagging secrets or 0 if they've never been used
```

Comment:
Would it not be simpler to return (and store) the expected next index? We never care about the last one anyway, we only care about the next one (which we use as a sender, or search for as a recipient). And the default value of 0 is a natural fit for the expected next one when we haven't seen any.

Reply:
It's a matter of semantics really. We could do it, just please on an upcoming refactor/cleanup PR!